### PR TITLE
Added a boolean in SolApplication class (Fix for issue #251)

### PR DESCRIPTION
**Pull request by manas96**
# Description
A fix for issue #251 .
Added boolean "isNewGame" in SolApplication class.
It is set to TRUE when a new game is made, and FALSE when previous game
is continued.
If TRUE, mercenaries.json file is not loaded. If FALSE, mercenaries.json is loaded.

# Testing
Launch new game after hiring mercenaries in a previous session. Mercenaries won't spawn in the new game now.

### CODE REVIEW

**Comment by GooeyHub** [comment]
Can one of the admins please verify this patch?

File: engine/src/main/java/org/destinationsol/SolApplication.java [hunk 1]
```diff
@@ -232,7 +234,9 @@ public OggMusicManager getMusicManager() {
         return musicManager;
     }
 
-    public OggSoundManager getSoundManager() {
-        return soundManager;
-    }
+    public OggSoundManager getSoundManager() { return soundManager; }
```
Comment (NicholasBatesNZ) [inline engine/src/main/java/org/destinationsol/SolApplication.java]: Why have you changed the formatting? Please revert

File: engine/src/main/java/org/destinationsol/SolApplication.java [hunk 2]
```diff
@@ -232,7 +234,9 @@ public OggMusicManager getMusicManager() {
...
+
+    public boolean getisNewGame() {return isNewGame;}
```
Comment (NicholasBatesNZ) [inline engine/src/main/java/org/destinationsol/SolApplication.java]: Same here, revert formatting please

File: engine/src/main/java/org/destinationsol/SolApplication.java [hunk 3]
```diff
@@ -232,7 +234,9 @@ public OggMusicManager getMusicManager() {
...
+
+    public void setisNewGame(boolean value) {isNewGame = value;}
```
Comment (NicholasBatesNZ) [inline engine/src/main/java/org/destinationsol/SolApplication.java]: And here :)

File: engine/src/main/java/org/destinationsol/game/SolGame.java [hunk 4]
```diff
@@ -236,6 +236,9 @@ private void createMercs() {
         if (!SaveManager.resourceExists(MERC_SAVE_FILE)) {
             return;
         }
+        if(solApplication.getisNewGame()){
```
Comment (NicholasBatesNZ) [inline engine/src/main/java/org/destinationsol/game/SolGame.java]: `if (...)` instead of `if(...)` (Add a space)

**Comment by Cervator** [comment]
Hey there thanks for the good report and quick fix :-)

Is your IDE just auto-formatting stuff away from the more common project formats, perhaps? Try to set up and run Checkstyle. There should be a config file for it after you've run just about anything Gradle, but if you can't spot it run `gradlew extractConfig`

Triggering bot for review

@GooeyHub: add to whitelist

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by manas96** [comment]
Fixed formatting stuff suggested by @NicholasBatesNZ .
@Cervator - seems I didn't have CheckStyle installed in my IntelliJ. When installing it, I got an error similar to [this ](https://github.com/MovingBlocks/Terasology/issues/3113)issue. 
I think the DestSol gradle config needs to be updated with the same fix. (I fixed my issue by deleting the offending module in my local copy)

**Comment by manas96** [comment]
I see your point, @vampcat . However, if I simply pass the variable through all invocations I will eventually have to know the value of `isNewGame` inside
` @Override
void updateCustom(...)` method of the `LoadingScreen` class. I can't simply pass in the boolean to this method, since it is overriding the `solUiScreen` interface method.
So I think the best solution is to move `isNewGame` inside the `LoadingScreen` class.
 If I understand this code correctly, this is the same reason why the `boolean loadTutorial` variable is also present in the `LoadingScreen` class.

**Comment by vampcat** [comment]
I am fine with shifting the boolean to LoadingScreen class. I would like to remove these variables in the future, but since that requires a slightly more complicated refactor.. :D

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by Steampunkery** [comment]
Hmm. In the upcoming galaxy generation PR, there is an isNewGame variable. Do we hold off on this until then or just fix the merge conflicts when we merge galaxy generation?

**Comment by vampcat** [comment]
Which class is that variable in?

**Comment by manas96** [comment]
It's in the SolGame class in the Galaxy generation PR.

**Pull request merged**